### ISSUE TITLE
Maxiter hard coded to 200,000; dlc 2.2b8

### ISSUE THREAD

**Issue by catubc** [DeepLabCut/DeepLabCut#1008]
Hello
I noticed that I can't change/update the max-iterations for multi-animal training beyond 200,000.
Is this intended as a feature or is it a bug?
Thanks so much,

                           [2, 4],
                             [6, 10],
                             [1, 6],
                             [1, 2]],
 'partaffinityfield_predict': True,
 'pos_dist_thresh': 17,
 'project_path': '/mnt/adfe6e7b-b77b-4731-bc9e-e639667faba4/lisa/cohort2-danlab-2020-10-21',
 'regularize': False,
 'rotation': 25,
 'rotratio': 0.4,
 'save_iters': 10000,
 'scale_jitter_lo': 0.5,
 'scale_jitter_up': 1.25,
 'scoremap_dir': 'test',
 'shuffle': True,
 'snapshot_prefix': '/mnt/adfe6e7b-b77b-4731-bc9e-e639667faba4/lisa/cohort2-danlab-2020-10-21/dlc-models/iteration-0/cohort2Oct21-trainset95shuffle1/train/snapshot',
 'stride': 8.0,
 'weigh_negatives': False,
 'weigh_only_present_joints': False,
 'weigh_part_predictions': False,
 'weight_decay': 0.0001}
Activating limb prediction...
Starting with multi-animal imaug + adam pose-dataset loader.
Batch Size is 8
Getting specs multi-animal-imgaug 27 11
Initializing ResNet
Loading ImageNet-pretrained resnet_50
2020-11-24 07:09:03.744053: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX
2020-11-24 07:09:03.771400: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 3391720000 Hz
2020-11-24 07:09:03.773384: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x55afbd32ae00 executing computations on platform Host. Devices:
2020-11-24 07:09:03.773408: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
2020-11-24 07:09:03.947429: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1433] Found device 0 with properties: 
name: TITAN Xp major: 6 minor: 1 memoryClockRate(GHz): 1.582
pciBusID: 0000:03:00.0
totalMemory: 11.91GiB freeMemory: 8.41GiB
2020-11-24 07:09:03.947477: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0
2020-11-24 07:09:03.948233: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-11-24 07:09:03.948250: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0 
2020-11-24 07:09:03.948259: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N 
2020-11-24 07:09:03.948508: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 8184 MB memory) -> physical GPU (device: 0, name: TITAN Xp, pci bus id: 0000:03:00.0, compute capability: 6.1)
2020-11-24 07:09:03.950205: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x55afbf310ee0 executing computations on platform CUDA. Devices:
2020-11-24 07:09:03.950237: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): TITAN Xp, Compute Capability 6.1
Max_iters overwritten as 200000
Display_iters overwritten as 1000
Save_iters overwritten as 50000
Training parameters:
{'stride': 8.0, 'weigh_part_predictions': False, 'weigh_negatives': False, 'fg_fraction': 0.25, 'mean_pixel': [123.68, 116.779, 103.939], 'shuffle': True, 'snapshot_prefix': '/mnt/adfe6e7b-b77b-4731-bc9e-e639667faba4/lisa/cohort2-danlab-2020-10-21/dlc-models/iteration-0/cohort2Oct21-trainset95shuffle1/train/snapshot', 'log_dir': 'log', 'global_scale': 0.8, 'location_refinement': True, 'locref_stdev': 7.2801, 'locref_loss_weight': 0.05, 'locref_huber_loss': True, 'optimizer': 'adam', 'intermediate_supervision': False, 'intermediate_supervision_layer': 12, 'regularize': False, 'weight_decay': 0.0001, 'crop_pad': 0, 'scoremap_dir': 'test', 'batch_size': 8, 'dataset_type': 'multi-animal-imgaug', 'deterministic': False, 'mirror': False, 'pairwise_huber_loss': False, 'weigh_only_present_joints': False, 'partaffinityfield_predict': True, 'pairwise_predict': True, 'all_joints': [[0], [1], [2], [3], [4], [5], [6], [7], [8], [9], [10]], 'all_joints_names': ['nose', 'leftear', 'rightear', 'spine1', 'spine2', 'spine3', 'spine4', 'spine5', 'tail1', 'tail2', 'tail3'], 'cropratio': 0.4, 'dataset': 'training-datasets/iteration-0/UnaugmentedDataSet_cohort2Oct21/cohort2_danlab95shuffle1.pickle', 'display_iters': 500, 'init_weights': '/home/cat/anaconda3/envs/DLC-GPU2/lib/python3.7/site-packages/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt', 'max_input_size': 1500, 'metadataset': 'training-datasets/iteration-0/UnaugmentedDataSet_cohort2Oct21/Documentation_data-cohort2_95shuffle1.pickle', 'min_input_size': 64, 'multi_step': [[0.0001, 7500], [5e-05, 12000], [1e-05, 200000]], 'net_type': 'resnet_50', 'num_joints': 11, 'num_limbs': 27, 'pafwidth': 20, 'pairwise_loss_weight': 0.1, 'partaffinityfield_graph': [[2, 5], [8, 9], [2, 6], [3, 4], [0, 5], [6, 7], [5, 9], [1, 5], [0, 1], [4, 5], [2, 7], [6, 8], [8, 10], [1, 4], [0, 2], [2, 8], [2, 3], [6, 9], [7, 8], [0, 3], [9, 10], [1, 3], [5, 6], [2, 4], [6, 10], [1, 6], [1, 2]], 'pos_dist_thresh': 17, 'project_path': '/mnt/adfe6e7b-b77b-4731-bc9e-e639667faba4/lisa/cohort2-danlab-2020-10-21', 'rotation': 25, 'rotratio': 0.4, 'save_iters': 10000, 'scale_jitter_lo': 0.5, 'scale_jitter_up': 1.25}
Starting multi-animal training....

**Comment by MMathisLab** [comment]
it uses batch training; it significantly overfits and affects performance if you go beyond 100K. (Also, I don't see in your example where you try to set it above 200,000 anyhow)

**Comment by AlexEMG** [comment]
But feel free to train as long as you see fit (on the test set).

**Comment by catubc** [comment]
Hi thanks for the response.  I don't quite understand how to change the 200,000 training limit though, it is hard-set for now.

Also I am not sure if overtraining occurs with all dataset types, we have quite messy and complex data (and are trying to train on tens of thousands of frames).

Thanks for the helpl.

**Comment by jacob-amme** [comment]
@AlexEMG, I have the same question as @catubc. Putting aside the question of whether or not overtraining occurs for maDLC models when trained beyond 200,000 iterations, how do I change this 200,000 training limit? In `deeplabcut.train_network` I tried setting `maxiters=300000` but it was still overwritten back to 200,000. To be more specific to my exact situation, I've successfully trained my maDLC model for 200,000 iterations but now I want to re-train it on the same training set for 100,000 more iterations. In the train>pose_cfg.yaml file, I changed `init_weights` to point to my last snapshot (snapshot-200000). However, my re-training job seems to not want to run because that 200,000 iteration threshold was reached in my first training session. Is there a way to overcome this?

**Comment by backyardbiomech** [comment]
You have to change the last multi-step, too. That also sets max_iter

**Comment by jacob-amme** [comment]
Thanks, @backyardbiomech! In the train>pose_cfg.yaml file I changed the last line of `multi_step` from `200000` (the default) to `300000`. My re-training job then ran for 100,000 more iterations, as intended. 

My assumption that for `deeplabcut.train_network`, setting `maxiters` to a certain value would overwrite whatever was set in the train>pose_cfg.yaml file came from the maDLC documentation, which says: 

`maxiters: This sets how many iterations to train. This variable is set in pose_config.yaml. However, you can overwrite it with this. If None, the value from there is used, otherwise it is overwritten! Default: None`

Since evidently, setting the `maxiters` does NOT actually overwrite what `multi_step` is set to in the train>pose_cfg.yaml file, I suggest that the maDLC documentation is amended to reflect this.
